lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -249,6 +249,10 @@ class ImageElements extends Gatherer {
   }
 
   /**
+   * Images might be sized via CSS. In order to compute unsized-images failures, we need to collect
+   * matched CSS rules to see if this is the case.
+   * Perf warning: Running this for 700 elements takes 1s to 5s.
```

Comment:
Just checking: did you mean 50 elements here (the previous limit)?

> 3s is currently the ~85th percentile on LR staging.

I presume that is 3s for the top 50, so it is surprising that 700 would only take 1-5s.

Reply:
yeah thx for bringing attention to these numbers.

> Just checking: did you mean 50 elements here (the previous limit)?

nah i was just trying to provide some general context. though TBH the count of total imageElements doesn't correlate too well. kohls has 700 and takes 1s on my machine and 5s on LR... however like 500 of them are data uris and 495 of _those_ are reusing network resources. 
but the gatherer cost correlates with # of network-resource-associated non-css non-shadowdom images that we do these protocol calls for. (something like ~160 for kohls i think? eh.)  

I'll remove this particular sentence and replace it with something more useful.

> > 3s is currently the ~85th percentile on LR staging.
> 
> I presume that is 3s for the top 50, so it is surprising that 700 would only take 1-5s.

this 3s 85th percentile figure is accurate for staging LR, which has the top50 filter for fetchElementWithSizeInformation but not for fetchSourceRules. 

700 does sound like a lot, but language selectors w/ flag sprites and other lazyload image stuff, the number can get high pretty quickly. shrug. i couldn't tell you what the distribution of ImageElements.length is for LR, but this investigation does make me curious...